cms/src/main/java/uk/nhs/digital/arc/process/ManifestProcessor.java

```diff
@@ -0,0 +1,301 @@
+            }
+
+            String calculatedDocbase = getDocbaseUsingManifestLocationIfRequired(wrapper.getDocumentBase(), manifestFileData);
+
+            if (calculatedDocbase == null || wrapper.getPages() == null || wrapper.getPages().size() == 0) {
+                ProcessOutcome outcome;
+                if (calculatedDocbase == null) {
+                    outcome = new ProcessOutcome("Could not locate a docbase in the file: " + manifestFile, ProcessOutcome.ERROR);
+                } else {
+                    outcome = new ProcessOutcome("Could not locate any page sections in the file: " + manifestFile, ProcessOutcome.ERROR);
+                }
+                messageSummary.addIndividualProcessOutcome(outcome);
+            } else {
+                // FOr each page, we send it to the processing method and record the outcome
+                wrapper.getPages().stream().forEach(page -> {
+                    try {
+                        messageSummary.addIndividualProcessOutcome(loadAndProcessPage(calculatedDocbase, page, objectMapper));
+                        createdPages.add(reportRootFolder + "/" + page.getPageName());
+                    } catch (IOException e) {
+                        e.printStackTrace();
+                    }
+                });
+
+                // Remembering that we might be changing things when properly processing data, we need to adjust
+                // the contents of the folder and remove pages that we no longer need as well as removing
+                // files from S3 that are no longer referenced (since their referencing JCR nodes wil have long been deleted)
+                if (!preview) {
+                    removeOrphanedPages(existingPages, createdPages);
+                }
+            }
+        } else {
+            messageSummary.addIndividualProcessOutcome(new ProcessOutcome("Could not locate manifest file in the location: " + manifestFile,
+                ProcessOutcome.ERROR));
+        }
+        return messageSummary;
+    }
+
+    /**
+     * A document base can be supplied as a fully qualified location or as a relative location.
+     * If it's relative, we'll use teh manifest location as a root
+     *
+     * @param documentBase which may not bea fully qualified URL
+     * @param manifestFileData which will be a fully qualified location
+     * @return a fully qualified location for the docbase
+     */
+    private String getDocbaseUsingManifestLocationIfRequired(String documentBase, FilePathData manifestFileData) {
+        FilePathData docbaseData = new FilePathData(documentBase);
+
+        if (documentBase == null) {
+            return null;
+        }
+
+        // If not a S3 reference (or HTTP or HTTPS) then assume a relative location
+        if (!docbaseData.looksLikeAUrl()) {
+            String assumedDocBase = "s3://" + manifestFileData.getS3Bucketname() + "/" + manifestFileData.getFolderPathNoBucket() + "/" + documentBase;
```

Comment:
Is it not possible to just return this without created the assumedDocBase variable?